tests/credentials_tests.c

```diff
@@ -572,6 +572,8 @@ static int s_cached_credentials_provider_queued_async_test(struct aws_allocator
         0);
 
     aws_credentials_provider_release(cached_provider);
+    ASSERT_TRUE(aws_atomic_load_int(&(cached_provider->ref_count)) == 0);
```

Comment:
I think these can go away now as they're not going to be reliably true due to async shutdown.